src/spec-lang/expr_grammar.pegjs

```diff
@@ -37,6 +38,26 @@ Invariant =
     return new SProperty(type as AnnotationType, expr, label !== null ? label : undefined, location());
   }
 
+Range =
+    start: Expression __ "..."  __ end: Expression
+    {
+      return [start, end];
+    }
+    / expression: Expression {return expression;}
+
+
+For_All =
+  type: FORALL __ "(" __ itr_type: IntType __ iterator: Identifier __ IN __ range: Range __ ")" __ expr: Expression
+  {
+    if(Array.isArray(range)) {
+      const [start, end] = range;
+      return new SForAll(itr_type, iterator, expr, start, end, undefined, location());
+    }
+    else {
+      return new SForAll(itr_type, iterator, expr, undefined, undefined, range, location());
+    }
```

Comment:
We can reduce nesting here. On the other hand, I can do a full cleanup pass for the grammar in other PR (if you do not mind).